Production_scripts/Process_extracts/extract_care_home.R

```diff
@@ -0,0 +1,120 @@
+#####################################################
+# Care Home Extract
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input - Social Care Care Home Episodes
+# Description - Match on Client data and Costs data
+#####################################################
+
+
+# Load packages
+library(dplyr)
+library(createslf)
+library(phsmethods)
+library(lubridate)
+
+
+year <- check_year_format("1920")
+
+
+# Read in data---------------------------------------
+
+source_ch_data <- haven::read_sav(get_sc_ch_episodes_path(ext = "zsav")) %>%
+  # select episodes for FY
+  filter(is_date_in_year(record_keydate1, year) |
+    (record_keydate1 <= end_fy(year) & record_keydate2 >= start_fy(year) | is.na(record_keydate2))) %>%
+  # remove any episodes where the latest submission was before the current year
+  filter(convert_fyyear_to_year(year) > substr(sc_latest_submission, 1, 4))
+
+
+# Match on Client Data ---------------------------------------
+
+# read client data in
+client_data <- readr::read_rds(get_source_extract_path(year, type = "Client"))
+
+# match to client data
+matched_data <- source_ch_data %>%
+  left_join(client_data, by = c("sending_location", "social_care_id"))
+
+
+# Data Cleaning ---------------------------------------
+
+source_ch_clean <- matched_data %>%
+  # create variables
+  mutate(
+    year = year,
+    recid = "CH",
+    SMRType = "Care-Home"
+  ) %>%
+  # compute lca variable from sending_location
+  mutate(lca = convert_sending_location_to_lca(sending_location)) %>%
+  # bed days
+  # create dummy end where blank
+  mutate(dummy_discharge = if_else(is.na(record_keydate2), end_fy(year) + days(1), record_keydate2)) %>%
+  create_monthly_beddays(year, record_keydate1, dummy_discharge) %>%
+  # year stay
+  mutate(yearstay = rowSums(across(ends_with("_beddays")))) %>%
+  # total length of stay
+  mutate(stay = difftime(record_keydate2, record_keydate1, units = "days"))
```

Comment:
The variable `stay` is returning NA in the dataframe but this should return 366 like yearstay (2020 was a leap year). The missing end dates in `record_keydate2` might be causing the issue here?

Reply:
is there something we can replace `record_keydate2` with to provide the stay to have a value? or if `NA` use quarter end date?

Reply:
I have replace the `stay` to use the `dummy_discharge` rather than `record_keydate2` as this code computes a date where `record_keydate2` is `NA`

does this make sense? give out the required output?

Reply:
This looks like it's working to me now

Reply:
This is working now but is also producing negative stays where the dummy discharge looks like it is before admission. We think this may be due to the filter, i have tried playing around with this but i can't get it to work. @Moohan would you mind having another look please? After this issue, i think the branch is ready to merge

Reply:
@Jennit07 can you have a look at #305 please? I think it will be relevant and useful here. I did it for GP OoHs but I think it's useful for here and for Home Care.

You might want to add a few more tests (`expect_true` / `expect_false`) based on the examples you've seen here.